### PR TITLE
Simplify Grpc Channel implementation

### PR DESCRIPTION
**Pull request by rongrong**
### What changes are proposed in this pull request?

Simplify Grpc Channel implementation

### Why are the changes needed?

Previously there GrpcChannelKey and GrpcConnectionKey, also
GrpcChannel and GrpcConnection. The distinction between them are
not clear. It seems that GrpcConnectionKey is GrpcChannelKey + groupIndex
(connection key has a channel key). GrpcChannel is GrpcConnection +
authentication (channel has a connection). This logic is very confusing.
Simplify the logic by merging GrpcChannelKey and GrpcConnectionKey into
GrpcChannelKey, which is the equivalent of the previous GrpcConnectionKey.
Also merging GrpcChannel and GrpcConnection into GrpcChannel, which is
equivalent to previous GrpcChannel. Also renaming GrpcConnectionPool to
GrpcChannelPool. Removing grpc connection to reduce confusion.

### CODE REVIEW

**Comment by rongrong** [comment]
Only last commit. First commit is reviewed in #15751.

File: core/common/src/main/java/alluxio/grpc/GrpcChannel.java [hunk 1]
```diff
@@ -11,69 +11,140 @@
...
-   * @param connection the grpc connection
-   * @param authDriver nullable client-side authentication driver
+   * Creates a new connection object.
+   * @param channelKey gRPC channel key
+   * @param managedChannel the underlying gRPC {@link ManagedChannel}
+   */
+  protected GrpcChannel(GrpcChannelKey channelKey, ManagedChannel managedChannel) {
+    mChannelKey = Objects.requireNonNull(channelKey, "channelKey is null");
+    mManagedChannel = Objects.requireNonNull(managedChannel, "managedChannel is null");
+    mChannel = managedChannel;
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply core/common/src/main/java/alluxio/grpc/GrpcChannel.java:68]: do we really need to keep `managedChannel`?

File: core/common/src/main/java/alluxio/grpc/GrpcChannelPool.java [hunk 2]
```diff
@@ -302,15 +288,13 @@ private EventLoopGroup acquireNetworkEventLoop(GrpcChannelKey channelKey,
           conf.getInt(PropertyKey.Template.USER_NETWORK_NETTY_WORKER_THREADS
               .format(key.getPropertyCode()));
 
-      v = new CountingReference<>(
-          NettyUtils.createEventLoop(nettyChannelType, nettyWorkerThreadCount, String.format(
-              "alluxio-client-netty-event-loop-%s-%%d", key.name()), true),
-          1);
       LOG.debug(
           "Created a new event loop. NetworkGroup: {}. NettyChannelType: {}, NettyThreadCount: {}",
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/grpc/GrpcChannelPool.java:292]: Haven't created it?

**Review by LuQQiu — APPROVED** [review]
LGTM, left some minor comments

**Comment by rongrong** [comment]
alluxio-bot, merge this please

**Pull request merged**